python/nx-cugraph/nx_cugraph/convert_matrix.py

```diff
@@ -65,8 +77,14 @@ def from_pandas_edgelist(
         src_indices = cp.asarray(np_or_cp.searchsorted(nodes, src_array), index_dtype)
         dst_indices = cp.asarray(np_or_cp.searchsorted(nodes, dst_array), index_dtype)
     else:
-        src_indices = cp.array(src_array)
-        dst_indices = cp.array(dst_array)
+        if is_src_copied:
+            src_indices = src_array
+        else:
+            src_indices = cp.array(src_array)
```

Comment:
Maybe I'm missing something obvious, but couldn't the same behavior be achieved by just doing the following?
```suggestion
        src_indices = cp.array(src_array, copy=False)
```

Reply:
I think you meant:
```python
        src_indices = cp.array(src_array, copy=not is_src_copied)
```
which should work as expected.

I found if-else branches more clear to use here, and easy for us to do given that we already have the booleans around.

Reply:
And once again, if you didn't catch it from one of my previous comments in previous PRs, numpy 2 changed semantics:
https://numpy.org/doc/stable/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword
so I think it's okay to be extra clear here.

Reply:
Thanks for the explanation, but no, actually that snippet wasn't what I was thinking.  But after re-reading again I think there's some necessary side effects that I was missing, but I'd rather not assume I know what they are.  Let's chat offline and I'll update the comment afterwards.

Reply:
Thanks, code comments added. NetworkX doesn't share ownership with input objects, so I figure neither should we.